lib/health_monitor/monitor.rb

```diff
@@ -38,13 +41,21 @@ def provider_result(provider, request)
       status: STATUSES[:ok]
     }
   rescue => e
+    
     configuration.error_callback.call(e) if configuration.error_callback
-
-    {
-      name: provider.provider_name,
-      message: e.message,
-      status: STATUSES[:error]
-    }
+    if e.class.superclass == HealthWarning
+      {
```

Comment:
Please follow the same style as the rest of the code. You can see that that your PR fails Travis CI because of that (you can run `rake rubocop` to see all the errors).